src/c++/perf_analyzer/genai-perf/genai_perf/export_data/report_output.py

```diff
@@ -0,0 +1,95 @@
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from argparse import Namespace
+
+from genai_perf.exceptions import GenAIPerfException
+from genai_perf.export_data.data_exporter_factory import (
+    DataExporterFactory,
+    DataExporterType,
+)
+from genai_perf.export_data.data_exporter_interface import DataExporterInterface
+from genai_perf.llm_metrics import LLMProfileDataParser
+from genai_perf.parser import get_extra_inputs_as_dict
+
+
+class ReportOutput:
+    """
+    A class to orchestrate output generation.
+    """
+
+    def __init__(self, data_parser: LLMProfileDataParser, args: Namespace):
+        self.data_parser = data_parser
+        self.args = args
+
+        if args.concurrency:
+            self.infer_mode = "concurrency"
+            self.load_level = f"{args.concurrency}"
+        elif args.request_rate:
+            self.infer_mode = "request_rate"
+            self.load_level = f"{args.request_rate}"
+        else:
+            raise GenAIPerfException("No valid infer mode specified")
+
+        self.stats = self.data_parser.get_statistics(self.infer_mode, self.load_level)
+
+    def report_output(self) -> None:
+        factory = DataExporterFactory()
+        data_exporters = []
+        data_exporters.append(self._get_console_exporter(factory))
+        data_exporters.append(self._get_json_exporter(factory))
+        data_exporters.append(self._get_csv_exporter(factory))
+
+        for exporter in data_exporters:
+            exporter.export()
+
+    def _get_console_exporter(
+        self, factory: DataExporterFactory
+    ) -> DataExporterInterface:
+        config = {
+            "type": DataExporterType.CONSOLE,
+            "stats": self.stats.stats_dict,
+        }
+        return factory.create_data_exporter(config)
+
+    def _get_json_exporter(self, factory: DataExporterFactory) -> DataExporterInterface:
+        config = {
```

Comment:
wait, the config isn't even an existing object. You are just creating a dict to pass args into a function? Why not just pass them as keyword args?

Reply:
Your suggestion was my first implementation.
I went with the map approach I have right now because once we get to a point where we try to support plugins, i couldn't understand how to make your suggested method work correctly.
Perhaps this is putting the cart 2 miles before the horse but they felt functional equivalent to me where this method is easier to extend later. While doing any refactoring and clean up work, I have been told to keep modularity in mind to set up us for the work coming.

Reply:
Yeah I'm not sure the clean way to do plugins. 
I don't know the best practices here, so I don't have any remaining comments. This can be resolved unless Hyunjae has strong opinions

Reply:
I'll resolve this comment since I don't have much strong preference here as well.